server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulator.java

```diff
@@ -240,8 +199,14 @@ private String getSectionName(Node node) {
   }
 
   private SymbolTable createOrGetSymbolTable(ProgramNode program) {
-    return programSymbols.computeIfAbsent(
-        SymbolTable.generateKey(program), p -> new SymbolTable());
+    String key = SymbolTable.generateKey(program);
+    synchronized (this.programSymbols) {
```

Comment:
Why does it have to synchronized?

Reply:
Also the new version of the code can be used with the `computeIfAbsent`.

Reply:
computeIfAbsent will cause concurrent modification of programSymbols map

Reply:
I am not sure what is going on here...
```
    this.programSymbols = Collections.synchronizedMap(new HashMap<>());
```
you added an explicit synchronized block and effectively emulated computeIfAbsent.

and all of that on a variable that as far as I can tell is local to the parsing thread until the processing is done.

Reply:
We were however running [this]( https://github.com/openjdk/jdk/blob/9c430c92257739730155df05f340fe144fd24098/src/java.base/share/classes/java/util/Collections.java#L2991).